boa/src/builtins/regexp/mod.rs

```diff
@@ -1366,10 +1464,10 @@ impl RegExp {
         let result = Self::abstract_exec(this, arg_str, context)?;
```

Comment:
Looking at `RegExpExec` spec, it can only return null or object. We can leverage the rust type system on this and use `Option<GcObject>`